### PR TITLE
Add PostgresVersion type (#183)

### PR DESCRIPTION
**Pull request by annymsMthd**
Signed-off-by: Joshua Benjamin <joshuabnjmn@gmail.com>

### CODE REVIEW

File: apis/kubedb/v1alpha1/postgres_version_helpers.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import (
+	crdutils "github.com/appscode/kutil/apiextensions/v1beta1"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+func (p PostgresVersion) OffshootName() string {
+	return p.Name
+}
+
+func (p PostgresVersion) OffshootLabels() map[string]string {
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha1/postgres_version_helpers.go]: This is not needed for `PostgresVersion`.

File: apis/kubedb/v1alpha1/postgres_version_helpers.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package v1alpha1
+
+import (
+	crdutils "github.com/appscode/kutil/apiextensions/v1beta1"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+func (p PostgresVersion) OffshootName() string {
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha1/postgres_version_helpers.go]: This is not needed for `PostgresVersion`.

File: apis/kubedb/v1alpha1/postgres_version_types.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	Spec              PostgresSpec   `json:"spec,omitempty"`
+	Status            PostgresStatus `json:"status,omitempty"`
+}
+
+// PostgresVersionSpec is the spec for postgres version
+type PostgresVersionSpec struct {
+	// Version
+	Version string `json:"version,omitempty"`
+	// Database Image
+	DB *PostgresVersionDatabase `json:"db,omitempty"`
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha1/postgres_version_types.go]: Not a pointer, since it is a required field.

**Comment by tamalsaha** [comment]
xref: https://github.com/kubedb/project/issues/183#issuecomment-409715800

File: apis/kubedb/v1alpha1/postgres_version_types.go (hunk 3 above)
Comment (annymsMthd) [reply apis/kubedb/v1alpha1/postgres_version_types.go]: Is this also the case for Exporter and Tools?

Comment (tamalsaha) [reply apis/kubedb/v1alpha1/postgres_version_types.go]: Yes.

**Pull request merged**